### DISCUSSION TITLE
Modal action with form and data

### DISCUSSION THREAD

**Discussion by HomaEEE** [filamentphp/filament#3014] [Help] [answered]
Hello, can I make button in row for modal form (Green button):
<img width="1065" alt="Снимок экрана 2022-07-03 в 01 58 42" src="https://user-images.githubusercontent.com/1429343/177018310-e142894e-fa1a-4cb1-91da-731f69616020.png">
and show custom modal with form - bill and payment type (photoshop):
![Untitled](https://user-images.githubusercontent.com/1429343/177018455-8cebafd1-43e9-4a38-a6c6-7cc63d340dc0.jpg)

_Current code of button_
`  ->prependActions([
                ButtonAction::make('paid')->label(__('filament/fields.button_pay'))
                ->icon('heroicon-s-cash')
                ->color('success')
                ->modalHeading('Оплата заказа')
                ->requiresConfirmation()
                ->action(fn () => $this->record->paid())
                ->modalContent(view('filament.events.actions.paid'))
                ->modalButton(__('filament/buttons.button_pay'))
            ])`

**Comment by danharrin** [comment, marked as answer]
Pass a `->form([components])` into the action, instead of `modalContent()`. You can access the form data using the `$data` parameter of `->action()`

**Reply by HomaEEE** [reply]
tanks, and may be you know - can create notify or toast with sound?
